File: apps/api/app/schemas/household.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class HouseholdBase(BaseModel):
    household_size: int = Field(ge=1, le=10)
    has_pet: bool = False
    pet_type: str | None = None
    has_baby: bool = False
    storage_level: Literal["small", "medium", "large"] = "medium"
    price_sensitivity: Literal["low", "medium", "high"] = "medium"
    stock_style: Literal["light", "normal", "heavy"] = "normal"
    preferred_platforms: list[str] | None = None
    reminder_tolerance: Literal["low", "medium", "high"] | None = None


class HouseholdCreate(HouseholdBase):
    pass


class HouseholdUpdate(HouseholdBase):
    pass


class HouseholdResponse(HouseholdBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    owner_account_id: str

File: packages/shared-types/src/household.ts
export type StorageLevel = "small" | "medium" | "large";
export type PriceSensitivity = "low" | "medium" | "high";
export type StockStyle = "light" | "normal" | "heavy";

export type HouseholdFormValues = {
  householdSize: number;
  hasPet: boolean;
  petType: string;
  hasBaby: boolean;
  storageLevel: StorageLevel;
  priceSensitivity: PriceSensitivity;
  stockStyle: StockStyle;
  preferredPlatforms?: string[];
  reminderTolerance?: "low" | "medium" | "high";
};
